### ISSUE TITLE
Default access point if WiFi is not configured

### ISSUE THREAD

**Issue by rp3tya** [motwok/mo2000_esphome_issues#2]
I am experimenting with WIFI_Now and made this setup:
- an ESP32 board with ethernet, running ESPHome with WIFI_Now (as gateway)
- two ESP8266 relay boards running ESPHome with WIFI_Now (as remote switches)

On ESP8266 (intentionally) no WiFi is configured because the relays are toggled via WIFI_Now, but there is a default access point always provided by them which can not be turned off (ssid ESP_xxxxxx, no password). If I configure a secure AP in ESPHome, then it replaces the default one, though mandatory WiFi connection/AP kind of defeats the purpose of WIFI_Now.

Thinking one step further, wouldn't it be great to have a "wifi_now switch" that can toggle an access point on ESP8266s? Then devices would communicate only via WIFI_Now (no WiFi connection nor AP) and when necessary, the AP could be turned on remotely, for example to do an OTA update.

**Comment by motwok** [comment]
first of all: You mix of ESP32 and ESP8266, does this work? (unfortunately i have problems with this mix specialy when Wifi is on a ESP32)

About the ESP_XXX accesspoint: i never saw this, nor did i know about this, please provide me your Versions of HA, ESPHome and the configuration you use.

About the future: Nope, i planed a kind of proxy that can handle api and ota-updates.
But feel free to use/implement a toogle for Wifi.

Emmo

**Comment by rp3tya** [comment]
Yes, ESP32 and ESP8266 can communicate with WIFI_Now, though `wifi` component is not configured on any of them. I also encountered issues with certain payload types or combinations, but for my experimentations I just used `int` which works fine.

ESP_XXX: it took me some time to find a pattern here, because I could not reproduce right away with a D1 Mini. Then I uploaded the same firmware to a Sonoff Mini (which is actually an ESP8285) and the AP appeared. I'm not even sure what this means...

But yeah, my goal would be to have devices communicate exclusively via ESP-Now and one of them act as a gateway (to HA via ethernet).

HA: 2021.4.5
ESPHome: 1.16.2
WIFI_Now: mid-March
Config: [wn2.yaml](https://github.com/motwok/mo2000_esphome_issues/files/6360426/wn2.yaml.txt)

**Comment by motwok** [comment]
> Yes, ESP32 and ESP8266 can communicate with WIFI_Now, though `wifi` component is not configured on any of them. I also encountered issues with certain payload types or combinations, but for my experimentations I just used `int` which works fine.

You are the second one who mention this, Andreas Spiess pointet me out that he had used strings to get esp_now to work in his projects.
Can you tell me which payload types fail in your setups?

> ESP_XXX: it took me some time to find a pattern here, because I could not reproduce right away with a D1 Mini. Then I uploaded the same firmware to a Sonoff Mini (which is actually an ESP8285) and the AP appeared. I'm not even sure what this means...

One little thing... when using WIFI then my implementation uses the channel set by WIFI... i havenot enougth insides of WIFI to fix this problem.
 
> But yeah, my goal would be to have devices communicate exclusively via ESP-Now and one of them act as a gateway (to HA via ethernet).

Emmo

**Comment by rp3tya** [comment]
> Can you tell me which payload types fail in your setups?

Actually I only had success with single `int` or `bool`. If I added two `int`s for example, on the receiver end the value of the second one was equal to the first one. Tested on two D1 Minis. Couldn't get strings to work either.

**Comment by motwok** [comment]
> ... If I added two `int`s for example, on the receiver end the value of the second one was equal to the first one. ...
Looks like a bug... i will check please provide me your yaml files.

>...Tested on two D1 Minis. Couldn't get strings to work either.
Are you testing on ESP8266 only?

**Comment by rp3tya** [comment]
I still had them open : )

![wifi_now](https://user-images.githubusercontent.com/14825453/120618150-75144080-c463-11eb-85ae-58d3c1452793.png)

And yes, both sender and receiver is ESP8266.
